localstack-core/localstack/utils/catalog/catalog.py

```diff
@@ -0,0 +1,140 @@
+import logging
+from abc import abstractmethod
+from enum import StrEnum
+
+from plux import Plugin
+
+from localstack.services.cloudformation.resource_provider import (
+    plugin_manager as cfn_plugin_manager,
+)
+from localstack.utils.catalog.common import (
+    AwsServiceOperationsSupportInLatest,
+    AwsServicesSupportInLatest,
+    CloudFormationResourcesSupportAtRuntime,
+    CloudFormationResourcesSupportInLatest,
+)
+from localstack.utils.catalog.loader import RemoteCatalogLoader
+
+ServiceName = str
+ServiceOperations = set[str]
+ProviderName = str
+CfnResourceName = str
+CfnResourceMethodName = str
+AwsServicesSupportStatus = AwsServicesSupportInLatest | AwsServiceOperationsSupportInLatest
+CfnResourceSupportStatus = (
+    CloudFormationResourcesSupportInLatest | CloudFormationResourcesSupportAtRuntime
+)
+
+LOG = logging.getLogger(__name__)
+
+
+class LocalstackEmulatorType(StrEnum):
+    COMMUNITY = "community"
+    PRO = "pro"
+
+
+class CatalogPlugin(Plugin):
+    namespace = "localstack.utils.catalog"
+
+    @staticmethod
+    def get_cfn_resources_catalog(cloudformation_resources: dict):
+        cfn_resources_catalog = {}
+        for emulator_type in list(LocalstackEmulatorType):
+            if emulator_type in cloudformation_resources:
+                for resource_name in cloudformation_resources[emulator_type].keys():
+                    cfn_resources_catalog[emulator_type] = {
+                        resource_name: set(
+                            cloudformation_resources[emulator_type][resource_name]["methods"]
+                        )
+                    }
+        return cfn_resources_catalog
```

Comment:
question: Is this to make sure that we only allow `community` and `pro` as the types?

Otherwise we could just simplify to:


```suggestion
    def get_cfn_resources_catalog(cloudformation_resources: dict):
        cfn_resources_catalog = {}
        for emulator_type, emulator in cloudformation_resources.items():
            for resource_name, resource in emulator.items():
                cfn_resources_catalog[emulator_type] = {
                    resource_name: set(
                            resource["methods"]
                    )
                }
        return cfn_resources_catalog
```

and lose a bit of complexity here